### PR TITLE
Provide Microsoft.Extentions.FileProviders.Abstractions package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `Microsoft.Extentions.FileProviders.Abstractions` NuGet package readme.

See #92228

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@dotnet/area-extensions-filesystem please review this package readme PR before EOW so that we can get the change still ported into .NET 8.

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,69 @@
...
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        IFileProvider fileProvider = new CustomFileProvider();
+        // Use the custom file provider
+    }
+}
+```
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md:38]: Can we have a functional IFileProvider instead? e.g: https://github.com/dazinator/Dazinator.Extensions.FileProviders/blob/7c00e96b0b4a7e40342ea2abc1cf1d418be2b660/src/Dazinator.Extensions.FileProviders/InMemory/InMemoryFileProvider.cs

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md [hunk 2]
```diff
@@ -2,43 +2,69 @@
...
 
 The main types provided by this library are:
 
-* ``
-* ``
-* ``
+* `IFileProvider`
+* `IDirectoryContents`
+* `IFileInfo`
+* `NullFileProvider`
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md:49]: Use fully qualified names.

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md [hunk 3]
```diff
@@ -2,43 +2,69 @@
...
 
 ## Additional Documentation
 
 <!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->
 
-* [Conceptual documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/**LIBRARYNAME**/overview)
-* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/**LIBRARYNAME**)
+* [Conceptual documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/change-tokens)
+* [API documentation](https://learn.microsoft.com/nl-nl/dotnet/api/microsoft.extensions.fileproviders)
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md:56]:
```suggestion
* [Conceptual documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/file-providers)
* [Detect changes with change tokens](https://learn.microsoft.com/aspnet/core/fundamentals/change-tokens)
* [API documentation](https://learn.microsoft.com/dotnet/api/microsoft.extensions.fileproviders)
```

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md [hunk 4]
```diff
@@ -2,43 +2,69 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
-
+`Microsoft.Extensions.FileProviders.Abstractions` serves as the foundation for creating file providers in .NET, offering core abstractions to develop custom file providers.
+It enables fetching files from various sources.
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md:6]:
```suggestion
Serves as the foundation for creating file providers in .NET, offering core abstractions to develop custom file providers capable of fetching files from various sources.
```

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md (hunk 1 above)
Comment (Jozkee) [reply src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md]: Yes, I agree that's a large example for this document. Nvm my suggestion.

File: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md [hunk 5]
```diff
@@ -2,43 +2,71 @@
...
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        IFileProvider fileProvider = new CustomFileProvider();
+        // Use the custom file provider
+    }
+}
+```
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md:39]: How about we change this section to something similar to [`MS.Extensions.DependencyInjection.Abstractions`](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/PACKAGE.md#how-to-use)?

```md
This package is typically used with an implementation of file provider abstractions, such as `Microsoft.Extensions.FileProviders.Composite` and `Microsoft.Extensions.FileProviders.Physical`.
```

**Review by jozkee — APPROVED** [review]

**Pull request merged**